services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchRulesSpec.groovy

```diff
@@ -177,53 +252,50 @@ class SwitchRulesSpec extends BaseSpecification {
     }
 
     @Unroll
-    def "Able to delete default rule from a switch (delete-action=#data.deleteRulesAction)"() {
-        assumeFalse("Unable to run the test because an OF_ 12 switch has one broadcast rule as the default",
-                dstSwitch.ofVersion == "OF_12" && data.cookie != Cookie.VERIFICATION_BROADCAST_RULE_COOKIE)
-
-        given: "A switch with some flow rules installed"
-        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
-        flowHelper.addFlow(flow)
+    def "Able to delete default rule from an #sw.ofVersion switch (#sw.dpId, delete-action=#data.deleteRulesAction)"() {
+        assumeFalse("Unable to run the test because an OF_12 switch has one broadcast rule as the default",
+                sw.ofVersion == "OF_12" && data.cookie != Cookie.VERIFICATION_BROADCAST_RULE_COOKIE)
```

Comment:
I think this is fine and it's the way I do it too, but I just thought that it is more correct to filter these 'iterations' directly in data provider and use assumptions only for 'runtime' things that we cannot predict (like topology unsuitable for the test). Just thoughts for the future

Reply:
Yes, probably in the future we will fix it.